extractor/src/main/java/org/schabi/newpipe/extractor/stream/HlsAudioStream.java

```diff
@@ -0,0 +1,60 @@
+package org.schabi.newpipe.extractor.stream;
+
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+
+public class HlsAudioStream extends AudioStream implements RefreshableStream {

```

Comment:
I don't think it's a property of all HLS streams that they are refreshable. I think we have two alternative ways we can handle this instead:
- `RefreshableStream` could just be an interface that some `Stream`s can implement, then each `Service` would override `VideoStream/AudioStream` and also implement `RefreshableStream` if needed, and the player could check for refreshable streams using `instanceof`. This would require a lot of boilerplate though.
- Add a `(Audio/Video)Stream refresh() {}` method to the base `Stream` class which builds a new refreshed stream, and make it return `null` if refreshing is not available.